extensions/amp-video/0.1/video-cache.js

```diff
@@ -215,3 +198,58 @@ function getCacheUrlService(videoEl, ampdoc) {
     .installExtensionForDoc(ampdoc, 'amp-cache-url')
     .then(() => Services.cacheUrlServicePromiseForDoc(videoEl));
 }
+
+/**
+ * Fetch the sources for the given video element.
+ * @param {!Element} videoEl
+ * @param {!AmpDoc} ampdoc
+ * @return {!Promise<!Object>} JSON representing AMP's cached video sources.
+ */
+function requestCachedVideoSources(videoEl, ampdoc) {
+  const {win} = ampdoc;
+  if (shouldUseInlineVideoResponse(videoEl)) {
+    const inlineResponseEl = win.document.getElementById(
+      'amp-google-video-cache-response'
+    );
+    if (inlineResponseEl?.textContent) {
+      try {
+        const inlineResponseJson = JSON.parse(inlineResponseEl.textContent);
+        return Promise.resolve(inlineResponseJson);
```

Comment:
We'll have to add an extra check for a `sources` key because of the changes we made to the previously launched implementation that had video IDs as keys